src/renderer/api/endpoints/pods.api.ts

```diff
@@ -477,10 +467,7 @@ export class Pod extends WorkloadKubeObject {
   }
 
   getSelectedNodeOs() {
-    if (!this.spec.nodeSelector) return;
-    if (!this.spec.nodeSelector["kubernetes.io/os"] && !this.spec.nodeSelector["beta.kubernetes.io/os"]) return;
-
-    return this.spec.nodeSelector["kubernetes.io/os"] || this.spec.nodeSelector["beta.kubernetes.io/os"];
+    return this.spec.nodeSelector?.["kubernetes.io/os"] || this.spec.nodeSelector?.["beta.kubernetes.io/os"];
```

Comment:
more side refactoring. Shouldn't it be:
```suggestion
    return this.spec?.nodeSelector?.["kubernetes.io/os"] || this.spec?.nodeSelector?.["beta.kubernetes.io/os"];
```

Reply:
I think that this is fine to be in this PR since I am already touching the `pods.api.ts` file.

With regards to your comment. No, spec is not an optional field so there is no point conditionally checking it.

Reply:
Maybe add a test for this while touching this method? 🙂

Reply:
Sounds good